modules/renter/hostdb/scan.go

```diff
@@ -46,14 +46,54 @@ var (
 // The gofunc is created inside of this function to eliminate the burden of
 // needing to remember to call 'go addHostToScanPool'.
 func (hdb *HostDB) scanHostEntry(entry *hostEntry) {
+	// Add the entry to a waitlist, then check if any thread is currently
+	// emptying the waitlist. If not, spawn a thread to empty the waitlist.
+	hdb.scanList = append(hdb.scanList, entry)
+	if hdb.scanWait {
+		// Another thread is emptying the scan list, nothing to worry about.
+		return
+	}
+
+	// Nobody is emptying the scan list, volunteer.
+	if hdb.tg.Add() != nil {
+		// Hostdb is shutting down, don't spin up another thread.
+		return
+	}
+	hdb.scanWait = true
 	go func() {
-		hdb.scanPool <- entry
+		defer hdb.tg.Done()
+
+		for {
+			hdb.mu.Lock()
+			if len(hdb.scanList) == 0 {
+				// Scan list is empty, can exit. Let the world know that nobody
+				// is emptying the scan list anymore.
+				hdb.scanWait = false
+				hdb.mu.Unlock()
+				return
+			}
+			// Get the next host, shrink the scan list.
+			entry := hdb.scanList[0]
+			hdb.scanList = hdb.scanList[1:]
```

Comment:
I was considering it. I'm not sure how the slice-append algo works, but if it only doubles based on the current size of the slice (instead of total historic usage) it'll basically be allocating memory every time there's an append, getting no advantage from amortization.